### ISSUE TITLE
ninja error

### ISSUE THREAD

**Issue by linucis** [hello2all/gamma-ray#1]
$:~/poco/gamma-ray/release$ ninja
[1/20] Building CXX object CMakeFiles/engine.dir/cmake_pch.hxx.gch
FAILED: CMakeFiles/engine.dir/cmake_pch.hxx.gch
/usr/bin/c++ -DCPPREST_FORCE_HTTP_CLIENT_ASIO -DCPPREST_FORCE_HTTP_LISTENER_ASIO -DCPPREST_NO_SSL_LEAK_SUPPRESS -DENABLE_DOCTEST_IN_LIBRARY -DPOCO_ENABLE_CPP11 -DPOCO_ENABLE_CPP14 -DPOCO_HAVE_FD_EPOLL -DPOCO_OS_FAMILY_UNIX -DXML_DTD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_REENTRANT -D_THREAD_SAFE -D_XOPEN_SOURCE=500 -I. -I../include -I../external/doctest/doctest -I../external/websocketpp -Wall -Werror -Wfatal-errors -Wpedantic -Wextra -Wmissing-include-dirs -pthread -O3 -DNDEBUG -Wall -Werror -Wfatal-errors -Wpedantic -Wextra -Wmissing-include-dirs -pthread -Wno-switch-enum -std=c++1z -Winvalid-pch -x c++-header -include /home/ubuntu/poco/gamma-ray/release/CMakeFiles/engine.dir/cmake_pch.hxx -MD -MT CMakeFiles/engine.dir/cmake_pch.hxx.gch -MF CMakeFiles/engine.dir/cmake_pch.hxx.gch.d -o CMakeFiles/engine.dir/cmake_pch.hxx.gch -c CMakeFiles/engine.dir/cmake_pch.hxx.cxx
cc1plus: error: ../external/doctest/doctest: No such file or directory [-Werror=missing-include-dirs]
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
[2/20] Building CXX object CMakeFiles/main.dir/cmake_pch.hxx.gch
FAILED: CMakeFiles/main.dir/cmake_pch.hxx.gch
/usr/bin/c++ -DCPPREST_FORCE_HTTP_CLIENT_ASIO -DCPPREST_FORCE_HTTP_LISTENER_ASIO -DCPPREST_NO_SSL_LEAK_SUPPRESS -DENABLE_DOCTEST_IN_LIBRARY -DPOCO_ENABLE_CPP11 -DPOCO_ENABLE_CPP14 -DPOCO_HAVE_FD_EPOLL -DPOCO_OS_FAMILY_UNIX -DXML_DTD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -D_REENTRANT -D_THREAD_SAFE -D_XOPEN_SOURCE=500 -I. -I../include -I../external/doctest/doctest -I../external/websocketpp -Wall -Werror -Wfatal-errors -Wpedantic -Wextra -Wmissing-include-dirs -pthread -O3 -DNDEBUG -flto -fno-fat-lto-objects -Wall -Werror -Wfatal-errors -Wpedantic -Wextra -Wmissing-include-dirs -pthread -Wno-switch-enum -std=c++1z -Winvalid-pch -x c++-header -include /home/ubuntu/poco/gamma-ray/release/CMakeFiles/main.dir/cmake_pch.hxx -MD -MT CMakeFiles/main.dir/cmake_pch.hxx.gch -MF CMakeFiles/main.dir/cmake_pch.hxx.gch.d -o CMakeFiles/main.dir/cmake_pch.hxx.gch -c CMakeFiles/main.dir/cmake_pch.hxx.cxx
cc1plus: error: ../external/doctest/doctest: No such file or directory [-Werror=missing-include-dirs]
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.

**Comment by hello2all** [comment]
@linucis What is your linux distribution? It seems your cpprestsdk is not properly installed, can you install according to this page https://github.com/microsoft/cpprestsdk and try again?

**Comment by linucis** [comment]
@hello2all ubuntu 18.04. ill try cpprestsdk.

**Comment by linucis** [comment]
@hello2all cpprestsdk is installed properly. 
--> libcpprest-dev is already the newest version (2.10.2-6).

**Comment by hello2all** [comment]
OK, I'll try to re-create this bug latter.

**Comment by hello2all** [comment]
Ok, I've re-created this bug. It seems `git clone --recurse-submodules` did not correctly pull `doctest` submodule. We can manually pull it.

```bash
cd ./gamma-ray/external
git clone https://github.com/onqtam/doctest.git
```

then compile again there should be no more errors
```bash
cd .. & cd release
ninja
```

**Comment by hello2all** [comment]
fixed in PR #5
